srunner/scenarios/background_activity.py

```diff
@@ -1073,23 +1073,24 @@ def _update_road_sources(self, prev_ego_index):
             for lane_key in self._road_dict:
                 source = self._road_dict[lane_key]
 
-                last_location = CarlaDataProvider.get_location(source.actors[-1])
-                if last_location is None:
-                    continue
-                source_location = source.wp.transform.location
-                ego_location = self._ego_wp.transform.location
-
-                # Stop the source from being too close to the ego or last lane vehicle
-                actor_dist = max(0, last_location.distance(source_location) - self._spawn_dist)
-                ego_dist = max(0, ego_location.distance(source_location) - min_distance)
-                move_dist = min(actor_dist, ego_dist)
-
-                # Move the source forward if needed
-                if move_dist > 0:
-                    new_source_wps = source.wp.next(added_dist)
-                    if not new_source_wps:
+                if source.actors:
```

Comment:
I don't like this. Due to stopping this, we have to later on move it back to its correct position. I suggest just changing the next code such that if no actors are found, the source keeps moving at a specific distance from the ego.

Reply:
Changed: If there are no actors, set the last_location to the ego's location, so that the source waypoint will keep moving forward.